src/entries.rs

```diff
@@ -0,0 +1,126 @@
+use crate::caches::CACHES_ENDPOINT;
+use crate::request::InfinispanRequest;
+use crate::Infinispan;
+use http::header::AUTHORIZATION;
+use http::Request;
+use reqwest::Response;
+use std::time::Duration;
+
+const TTL_HEADER: &str = "timeToLiveSeconds";
+
+pub struct CreateEntryReq<'i> {
+    infinispan_client: &'i Infinispan,
+    cache_name: String,
+    entry_name: String,
+    value: Option<String>,
+    ttl: Option<Duration>,
+}
+
+impl<'i> CreateEntryReq<'i> {
+    pub fn new(
+        infinispan_client: &'i Infinispan,
+        cache_name: impl Into<String>,
+        entry_name: impl Into<String>,
+    ) -> CreateEntryReq<'i> {
+        CreateEntryReq {
+            infinispan_client,
+            cache_name: cache_name.into(),
+            entry_name: entry_name.into(),
+            value: None,
+            ttl: None,
+        }
+    }
+
+    pub fn with_value(mut self, value: String) -> CreateEntryReq<'i> {
+        self.value = Some(value);
+        self
+    }
+
+    pub fn with_ttl(mut self, ttl: Duration) -> CreateEntryReq<'i> {
+        self.ttl = Some(ttl);
+        self
+    }
+
+    pub async fn send(&self) -> Result<Response, reqwest::Error> {
```

Comment:
Do you plan to define your own error types and avoid exposing Reqwest to users of this API?

Reply:
Yes: https://github.com/3scale-labs/infinispan-rs/pull/1#discussion_r614897095